arkouda/segarray.py

```diff
@@ -115,12 +115,22 @@ def __init__(self, segments, values, copy=False, lengths=None, grouping=None):
         else:
             self.lengths = lengths
         self.dtype = values.dtype
+
+        self.non_empty = zeros(self.size, dtype=akbool)
+        self.non_empty_count = 0
```

Comment:
Do we want these to be visible to users? I'm thinking we should rename these to `self._non_empty` and `self._non_empty_count` if we are planning on them only being accessed by internal code

Reply:
Updated to `_non_empty` and `_non_empty_count`. I was not sure if they would have any user facing value, but rethinking it, I am not sure they would.